corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
+    def get(cls, doc_id):
```

Comment:
Actually some overlap with my frustrations in https://github.com/dimagi/commcare-hq/pull/7577#issuecomment-123734212, that could be solved by the approach I was taking there, plus this proposed mixin